ipam/ipam.go

```diff
@@ -146,26 +153,21 @@ func (ipam *IPAMSvc) legacyAllocateIpByName(input interface{}, ctx common.RestCo
 	for _, s := range segments {
 		if s.Name == segmentName {
 			found = true
-			vm.SegmentId = fmt.Sprintf("%d", s.Id)
+			Endpoint.SegmentId = fmt.Sprintf("%d", s.Id)
 			break
 		}
 	}
 	if !found {
 		return nil, errors.New("Segment with name " + hostName + " not found")
 	}
-	log.Printf("Sement name %s has ID %s", segmentName, vm.SegmentId)
+	log.Printf("Segment name %s has ID %s", segmentName, Endpoint.SegmentId)
 
-	return ipam.addVm(&vm, ctx)
+	return ipam.addEndpoint(&Endpoint, ctx)
 }
 
 // handleHost handles request for a specific host's info
-func (ipam *IPAMSvc) addVm(input interface{}, ctx common.RestContext) (interface{}, error) {
-	vm := input.(*Vm)
-	err := ipam.store.addVm(ipam.dc.EndpointSpaceBits, vm)
-	if err != nil {
-		log.Printf("HEYHEYHEY %v\n", err)
-		return nil, err
-	}
+func (ipam *IPAMSvc) addEndpoint(input interface{}, ctx common.RestContext) (interface{}, error) {
```

Comment:
We should have a few comments throughout this function. It's long and complex enough.